esti/gc_test.go

```diff
@@ -0,0 +1,217 @@
+			epochCommitDateInSeconds := currentEpochInSeconds - (dayInSeconds * branch.deleteCommitDaysAgo)
+			_, err = client.CommitWithResponse(ctx, repo, branch.name, &api.CommitParams{}, api.CommitJSONRequestBody{Message: "Deleted file" + testCase.id, Date: &epochCommitDateInSeconds})
+			if err != nil {
+				t.Fatal("Commit some data", err)
+			}
+			_, _ = uploadFileRandomData(ctx, t, repo, branch.name, "file"+testCase.id+"not_deleted", false)
+			// This is for the previous commit to be the HEAD of the branch outside the retention time (according to GC https://github.com/treeverse/lakeFS/issues/1932)
+			_, err = client.CommitWithResponse(ctx, repo, branch.name, &api.CommitParams{}, api.CommitJSONRequestBody{Message: "not deleted file commit: " + testCase.id, Date: &epochCommitDateInSeconds})
+			if err != nil {
+				t.Fatal("Commit some data", err)
+			}
+		} else {
+			_, err = client.DeleteBranchWithResponse(ctx, repo, branch.name)
+			if err != nil {
+				t.Fatal("Delete brach", err)
+			}
+		}
+	}
+	return commitId
+}
+
+func getGCMode(testCase testCase) int {
+	if testCase.testMode == "mark" {
+		return markOnlyMode
+	} else if testCase.testMode == "sweep" {
+		return sweepOnlyMode
+	} else {
+		return fullGCMode
+	}
+}
+
+func validateGCJob(t *testing.T, ctx context.Context, testCase testCase, existingRef string) {
+	repo := committedGCRepoName + testCase.id
+
+	res, _ := client.GetObjectWithResponse(ctx, repo, existingRef, &api.GetObjectParams{Path: "file" + testCase.id})
+	fileExists := res.StatusCode() == 200
+
+	if fileExists && testCase.fileDeleted {
+		t.Fatalf("Expected the file to be removed by the garbage collector but it has remained in the repository. Test case '%s'. Test description '%s'", testCase.id, testCase.description)
+	} else if !fileExists && !testCase.fileDeleted {
+		t.Fatalf("Expected the file to remain in the repository but it was removed by the garbage collector. Test case '%s'. Test description '%s'", testCase.id, testCase.description)
+	}
+	locations := []string{"not_deleted_file1", "not_deleted_file2", "not_deleted_file3"}
+	for _, location := range locations {
+		res, _ = client.GetObjectWithResponse(ctx, repo, "main", &api.GetObjectParams{Path: location})
+		if res.StatusCode() != 200 {
+			t.Fatalf("expected '%s' to exist. Test case '%s', Test description '%s'", location, testCase.id, testCase.description)
+		}
+	}
+}
+
+func testPostCommittedGC(t *testing.T) {
+	ctx := context.Background()
+
+	findingPath := filepath.Join(os.TempDir(), gcFindingsFilename)
+	bytes, err := os.ReadFile(findingPath)
+	if err != nil {
+		t.Fatalf("Failed to read '%s': %s", findingPath, err)
+	}
+	fileRefPerTestCase := make(map[string]string)
+
+	err = json.Unmarshal(bytes, &fileRefPerTestCase)
+	if err != nil {
+		t.Fatalf("Failed to unmarshal findings '%s': %s", findingPath, err)
+	}
+
+	for _, testCase := range testCases {
+		existingRef := fileRefPerTestCase[testCase.id]
+		validateGCJob(t, ctx, testCase, existingRef)
```

Comment:
This does not refer to the name of the failing test case, which will make things harder to understand on failure.  I believe that here too we need to use t.Run to run as a subtest:
```suggestion
		t.Run(fmt.Sprintf("Test case %s", testCase.id), func (t *testing.T) {
			existingRef := fileRefPerTestCase[testCase.id]
			validateGCJob(t, ctx, testCase, existingRef)
		}
```